cinema/views.py

```diff
@@ -48,6 +75,31 @@
     queryset = MovieSession.objects.all()
     serializer_class = MovieSessionSerializer
 
+    def get_queryset(self):
+        queryset = self.queryset
+
+        date = self.request.query_params.get("date")
+        movie = self.request.query_params.get("movie")
+
+        if date:
+            queryset = queryset.filter(show_time__date=date)
+
+        if movie:
+            movie_ids = [int(str_id) for str_id in movie.split(",")]
+            queryset = queryset.filter(movie__id__in=movie_ids)
+
+        if self.action == "list":
+            queryset = queryset.select_related(
+                "cinema_hall",
+                "movie",
+            ).annotate(
+                tickets_available=F("cinema_hall__rows")
+                * F("cinema_hall__seats_in_row")
+                - Count("tickets")
+            )
+
+        return queryset.distinct()
```

Comment:
In the `get_queryset` method of `MovieSessionViewSet`, you are modifying `self.queryset` directly. This can lead to unexpected behavior if the method is called multiple times. Consider using a local variable to store the filtered queryset and return it at the end of the method.